### DISCUSSION TITLE
關於永遠覺得自己數學不好的那件事

### DISCUSSION THREAD

**Discussion by HsuSungTing** [1101-GS4719-C/Python-Community-GS4719-C#1299] [General]
我從國中開始，就覺得自己數學不好，而且很沒有自信
到高中因為身邊都是強者，所以覺得自己數學更爛了
但我其實很好奇，數學到底要多好才可以當一個和個的工程師?以及數學好是要算題目很快嗎?還是要能解決很複雜的問題?

於是我問Chatgpt，對於最吃重數學的軟體工程師而言，數學到底要多好?

同一行業中的特定職位和專案會有所不同。在某些數學要求較高的領域中，例如密碼學、電腦圖形學或機器學習算法設計等，對數學的要求可能更為嚴格。

對於計算機科學或軟體工程師而言，以下是一些常見的衡量數學能力的標準：

熟悉代數和微積分：擁有良好的代數和微積分基礎是計算機科學領域的基本要求。您應該熟悉各種函數、方程、極限、導數和積分等基本概念，能夠應用它們解決實際問題。

理解離散數學：離散數學在計算機科學中扮演重要角色，包括集合論、圖論、邏輯、概率論和組合數學等。您需要掌握離散數學的基本概念和算法，以便能夠處理離散結構、優化問題和算法設計等。

掌握線性代數：線性代數在電腦圖形學、機器學習和數據分析等領域中扮演重要角色。了解矩陣、向量、線性變換和特徵值等概念，並能夠應用線性代數方法解決相關問題。

熟練運用概率和統計：概率和統計在數據科學、機器學習和人工智能等領域中非常重要。您需要熟悉概率分佈、假設檢驗、迴歸分析、統計推斷等，以便能夠分析和解釋數據、評估算法性能等。

解決複雜問題的能力：除了純粹的數學知識，數學能力也體現在您能夠運用數學思維解決複雜問題的能力上。這包括邏輯推理、抽象建模、算法分析和優化等能力。

**Comment by YoYooHuang** [comment]
我覺得數學不好不代表你這個人不好，我相信無論是誰，只要持續努力，總有一天能超越其他人！而現在會覺得表現不好應該就是在還債而已，因為自己以前的不努力，導致現在要花費更多心力來達到目標，加油，你一定可以的！

**Comment by orianna901001** [comment]
我之前遇到微積分不會算的時候，會去問身邊的理工科同學，這時候其實就可以看出我們彼此之間的數學天賦，但是我也認同上面的留言"數學不好不代表你這個人不好"，其實我們兩個人各有強跟弱的地方。

**Comment by Miinteaa** [comment]
以前在國高中時測性向測驗時，我的邏輯能力與數理能力完全是天壤之別，所以對我來說數學跟邏輯能力不能完全掛勾。只要遇到比較困難的題目腦袋就會當機，我覺得跟心理壓力也有相當的關連，希望我可以早日克服看到數學就會排斥的心理。
